python3/raygun4py/raygunprovider.py

```diff
@@ -295,42 +295,44 @@ def _create_message(
         )
 
     def _transform_message(self, message):
+        message = message.copy()
         message = utilities.ignore_exceptions(self.ignored_exceptions, message)
 
         if message is not None:
-            message = utilities.filter_keys(self.filtered_keys, message)
-            message["details"]["groupingKey"] = utilities.execute_grouping_key(
+            details = message.get_details()
+            details = utilities.filter_keys(self.filtered_keys, details)
```

Comment:
Pass the `details` to the `filter_keys` function instead of passing the whole `message` object